# Test VotingClassifier:

#### 1. Use the breast-bin.csv dataset

In [1]:
from si.io.csv_file import read_csv

In [2]:
df_path = r"C:\Users\Mariana\si\datasets\breast-bin\breast-bin.csv"
df = read_csv(df_path, sep = ',', features = True, label = True) 

#### 2. Split the data into train and test sets

In [3]:
from si.model_selection.split import train_test_split
from si.models.decision_tree_classifier import DecisionTreeClassifier

In [4]:
train_dataset, test_dataset = train_test_split(df, test_size=0.33, random_state=42)

In [5]:
print(f'(As dimensões dos dados de treino correspondem a:', train_dataset.shape())
print(f'(As dimensões dos dados de teste correspondem a:',test_dataset.shape())

(As dimensões dos dados de treino correspondem a: (468, 9)
(As dimensões dos dados de teste correspondem a: (230, 9)


Os dados de treino apresentam 468 linhas e 49 colunas, enquanto que os dados de teste apresentam 230 linhas e 9 colunas.

#### 3. Create a KNN Classifier model

In [6]:
from si.models.knn_classifier import KNNClassifier

In [7]:
knn_model = KNNClassifier(k=3)

#### 4. Create a Logistic Regression model

In [8]:
from si.models.logistic_regression import LogisticRegression

In [9]:
logistic_regression_model = LogisticRegression(l2_penalty=1, alpha=0.001, max_iter=1000)

#### 5. Create a Decision Tree model

In [10]:
from si.models.decision_tree_classifier import DecisionTreeClassifier

In [11]:
decision_tree_model = DecisionTreeClassifier(max_depth=5)

#### 6. Create a Voting Classifier ensemble using the previous models

In [12]:
from si.ensemble.voting_classifier import VotingClassifier

In [13]:
voting_classifier = VotingClassifier([knn_model, logistic_regression_model, decision_tree_model])

#### 7. Train the Voting Classifier. What is the score of the model on the test set?

In [14]:
voting_classifier.fit(train_dataset)

In [15]:
voting_classifier.predict(test_dataset)

array([1., 1., 0., 0., 0., 0., 1., 0., 0., 1., 0., 1., 1., 1., 0., 0., 0.,
       0., 1., 0., 0., 1., 0., 1., 1., 1., 0., 0., 0., 0., 1., 0., 1., 0.,
       0., 0., 0., 0., 1., 1., 0., 0., 1., 0., 0., 1., 1., 0., 0., 1., 1.,
       0., 1., 0., 0., 1., 0., 0., 1., 0., 1., 1., 1., 0., 0., 1., 0., 0.,
       1., 0., 0., 0., 0., 0., 1., 0., 0., 1., 0., 1., 0., 1., 0., 0., 0.,
       1., 1., 0., 0., 0., 0., 1., 0., 1., 1., 0., 0., 0., 0., 1., 1., 0.,
       1., 1., 0., 1., 0., 0., 0., 1., 1., 1., 0., 0., 0., 1., 0., 1., 1.,
       0., 0., 0., 1., 0., 0., 0., 0., 0., 0., 0., 0., 1., 0., 0., 0., 1.,
       0., 0., 1., 0., 1., 0., 1., 0., 0., 0., 0., 0., 0., 1., 0., 1., 1.,
       1., 0., 0., 0., 1., 1., 0., 0., 0., 1., 1., 0., 0., 0., 0., 0., 0.,
       0., 1., 0., 0., 1., 1., 0., 1., 1., 1., 1., 1., 1., 0., 1., 1., 0.,
       0., 0., 1., 1., 0., 1., 0., 0., 1., 0., 0., 0., 0., 0., 0., 1., 0.,
       0., 0., 0., 1., 0., 0., 0., 1., 0., 0., 1., 0., 1., 1., 1., 0., 1.,
       0., 0., 0., 0., 0.

In [16]:
score = voting_classifier.score(test_dataset)
print(f"Score: {score}")

Score: 0.9782608695652174


O score de 0.9783 indica que o modelo obteve uma precisão de cerca de 97.8% no conjunto de dados de teste. Este valor sugere que o modelo utilizado é bastante bom para a classificação no conjunto de teste.